tests/client_encode.rs

```diff
@@ -0,0 +1,169 @@
+mod client_encode {
+    use async_h1::client;
+    use async_std::io::Cursor;
+    use async_std::prelude::*;
+    use client::Encoder;
+    use http_types::Body;
+    use http_types::Result;
+    use http_types::{Method, Request, Url};
+    use pretty_assertions::assert_eq;
+
+    async fn encode_to_string(request: Request, len: usize) -> http_types::Result<String> {
+        let mut buf = vec![];
+        let mut encoder = Encoder::encode(request).await?;
+        loop {
+            let mut inner_buf = vec![0; len];
+            let bytes = encoder.read(&mut inner_buf).await?;
+            buf.extend_from_slice(&inner_buf[..bytes]);
+            if bytes == 0 {
+                return Ok(String::from_utf8(buf)?);
+            }
+        }
+    }
+
+    #[async_std::test]
+    async fn client_encode_request_add_date() -> Result<()> {
+        let url = Url::parse("http://localhost:8080").unwrap();
+        let mut req = Request::new(Method::Post, url);
+        req.set_body("hello");
+
+        assert_encoded(
+            100,
+            req,
+            r#"POST / HTTP/1.1
+host: localhost:8080
+content-length: 5
+content-type: text/plain;charset=utf-8
```

Comment:
headers did not have a stable order last I checked, has that changed?

Reply:
yes! not as of this pr, but these headers are stable because they're hardcoded in async-h1 as of this pr.  in #158, we're introducing a stable and predictable sort order (currently `HOST` first, then alphabetically). that logic will eventually move into the http-types iterators, hopefully, but i'm trying to do this in orderly steps so there isn't a whole graph of dependent versions

Reply:
(also these tests were already doing identical assertions, it was just hidden because the headers were in fixture files)